unionml/model.py

```diff
@@ -570,12 +619,19 @@ def predict(
                 "ModelArtifact not found. You must train a model first with the `train` method before generating "
                 "predictions."
             )
+
+        wf: Workflow
         if features is None:
-            return self.predict_workflow()(model_object=self.artifact.model_object, **reader_kwargs)
-        return self.predict_from_features_workflow()(
-            model_object=self.artifact.model_object,
-            features=self._dataset.get_features(features),
-        )
+            wf = self.predict_workflow()(model_object=self.artifact.model_object, **reader_kwargs)
+        else:
+            wf = self.predict_from_features_workflow()(
+                model_object=self.artifact.model_object,
+                features=self._dataset.get_features(features),
+            )
+
+        # TODO(review,zevisert): Should local predict run callbacks too?
+
+        return wf
```

Comment:
what's the purpose of this change?

re: the TODO question: I think for this PR we can have callbacks invoked everywhere. I'd imagine it would be useful for testing.

We can then think about ways in which we can suppress certain callbacks in different situations, esp. if we need subtleties around whether they should be called in a Flyte execution or how they should be called in a FastAPI app.

Reply:
the output of this function should be the result of invoking `wf` locally, so it doesn't actually return a workflow.